indico/modules/events/contributions/forms.py

```diff
@@ -22,18 +22,18 @@
 from wtforms.validators import DataRequired
 
 from indico.core.db.sqlalchemy.protection import ProtectionMode
-from indico.modules.events.contributions.fields import ContributionPersonListField
+from indico.modules.events.contributions.fields import ContributionPersonListField, SubContributionPersonListField
 from indico.web.forms.base import IndicoForm
 from indico.web.forms.fields import TimeDeltaField, PrincipalListField, IndicoEnumRadioField, IndicoLocationField
-from indico.web.forms.validators import UsedIf
+from indico.web.forms.validators import UsedIf, MaxDuration
 from indico.util.i18n import _
 
 
 class ContributionForm(IndicoForm):
     title = StringField(_("Title"), [DataRequired()])
     description = TextAreaField(_("Description"))
-    duration = TimeDeltaField(_("Duration"), [DataRequired()], default=timedelta(minutes=20),
-                              units=('minutes', 'hours'),
+    duration = TimeDeltaField(_("Duration"), [DataRequired(), MaxDuration(timedelta(hours=24))],
```

Comment:
But the `MaxDuration` is only there to check that the duration value is not more than 24 hours (that is what we did in the legacy code)